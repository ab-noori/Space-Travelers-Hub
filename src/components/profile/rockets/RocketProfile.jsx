import PropTypes from 'prop-types';

import './rocketProfile.scss';

const RocketProfile = ({ rocket }) => (
  <li className="rockets-reserved">
    {rocket.rocketName}
  </li>
);

RocketProfile.propTypes = {
  rocket: PropTypes.shape({
    rocketName: PropTypes.string,
    rocketDescription: PropTypes.string,
    reserved: PropTypes.bool,
    flickrImages: PropTypes.arrayOf(PropTypes.string),
    id: PropTypes.string,
  }).isRequired,
};

export default RocketProfile;
