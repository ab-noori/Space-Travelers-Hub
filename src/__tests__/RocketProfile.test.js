import React from 'react';
import renderer from 'react-test-renderer';
import RocketProfile from '../components/profile/rockets/RocketProfile';

describe('RocketProfile', () => {
  it('renders correctly', () => {
    const rocket = {
      rocketName: 'Falcon 9',
      rocketDescription: 'A two-stage rocket...',
      reserved: true,
      flickrImages: ['url1', 'url2'],
      id: 'rocket1',
    };

    const rocketProfile = renderer.create(<RocketProfile rocket={rocket} />).toJSON();

    expect(rocketProfile).toMatchSnapshot();
  });
});
