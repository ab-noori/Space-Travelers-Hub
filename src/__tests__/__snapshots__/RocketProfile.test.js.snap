// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketProfile renders correctly 1`] = `
<li
  className="rockets-reserved"
>
  Falcon 9
</li>
`;
